manifester/helpers.py

```diff
@@ -226,3 +235,139 @@ def __getitem__(self, key):
     def __call__(self, *args, **kwargs):
         """Allow MockStub to be used like a function."""
         return self
+
+
+class InvalidVaultURLForOIDC(Exception):
+    """Raised if the vault doesn't allow OIDC login."""
+
+
+class Vault:
+    """Helper class for retrieving secrets from HashiCorp Vault."""
+
+    HELP_TEXT = (
+        "The Vault CLI in not installed on this system."
+        "Please follow https://learn.hashicorp.com/tutorials/vault/getting-started-install to "
+        "install the Vault CLI."
+    )
+
+    def __init__(self, env_file=".env"):
+        manifester_directory = Path()
+
+        if "MANIFESTER_DIRECTORY" in os.environ:
+            envar_location = Path(os.environ["MANIFESTER_DIRECTORY"])
+            if envar_location.is_dir():
+                manifester_directory = envar_location
+        self.env_path = manifester_directory.joinpath(env_file)
```

Comment:
These would likely be best done in a common place, like the settings module

Reply:
@JacobCallahan This actually is done in the settings module, but I revisited this today and remembered why I had repeated it here: vault_login.py throws an error when trying to retrieve settings values if the user is not already logged in to Vault. If I'm understanding the situation correctly, vault_login.py imports helpers.py, in which it encounters an assignment that uses settings.get(), then dynaconf attempts to process the settings file, which includes a Vault-formatted setting (offline_token), and dynaconf subsequently tries to retrieve that value from Vault before the login process has completed. This is also why I had hard-coded the log level in this module, which you commented on as well.

There may be another way to work around this race condition, but I have not figured one out yet. Do you have any thoughts on an alternative, or alternatively, do you think my admittedly inelegant solution is acceptable?

Reply:
Hmmm not ideal, but if it needs to be repeated anyway for the vault login then it's up to you whether that repetition happens here or in the vault script. I have no objections either way now.